docker/Dockerfile

```diff
@@ -0,0 +1,36 @@
+# First stage: build the application
+FROM golang:1.13.0-alpine AS builder
+ARG GOARCH
+ARG GOARM
+
+WORKDIR /go/src/github.com/CESARBR/knot-babeltower
+COPY . .
+
+# Install build utilities and glide dependency manager
+RUN apk add --update --no-cache --virtual .build-deps \
+    make \
+    git \
+    curl
+RUN curl https://glide.sh/get | sh
+
+# Install project development tools and dependencies
+RUN go get github.com/ahmetb/govvv
+RUN make deps
+
+# Build the application
+RUN make bin
```

Comment:
Since we constrain the base for the application in the `Dockerfile`, we could further optimize the building process (when using containers) by compiling only for linux target and disabling cross compilation :smile: 

```suggestion
# Build the app for linux target (cross compile disabled)
RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -installsuffix cgo -o app cmd/main.go
```

Reply:
Actually, in the end, the `make bin` command will run the command as you suggested targeting the Linux OS. The difference is that we use a script to get the OS where the command is being executed. The cross-compilation is disabled since the golang compiler is smart enough to enable it only when compiling for non-Linux OS.

Just for sake of curiosity, I executed both commands and compared them:

Your suggestion:
```
real	0m27.889s
user	0m38.428s
sys	0m7.506s

```
Makefile:
```
real	0m28.716s
user	0m38.465s
sys	0m7.537s
```

That script was the cause of that 1 second of difference with respect to the `real` metric.

Reply:
I don't think it's worth changing (in this case) and that script seems to work fine even if you run the `Makefile` on Windows, for example. What is your opinion?

Reply:
The advantage of using Makefile: only one file to touch when we need to change something related to the compilation.

Reply:
> _Actually, in the end, the make bin will run the command as you suggested (targeting the Linux OS). The difference is that we use a script to get the OS in which the command is being executed. The cross-compilation is disabled since the golang compiler is smart enough to enable it only when compiling for non-Linux OS."_

Considering the comment above, I totally agree with you :smile: 
BTW, +1 for those metrics! Let me know how you did those, later! :confetti_ball: